### ISSUE TITLE
Camera freeze on QR code scanning to link new devices

### ISSUE THREAD

**Issue by manzerbredes** [signalapp/Signal-Android#10989]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When I try to link another device with my Galaxy Tab A7, the camera image is showing only for 1 seconds then it freezes. Thus I cannot link other devices.

### Steps to reproduce
- Open Signal
- Goto settings
- Link devices 

**Actual result:** Camera show an image for 1 second then freezes
**Expected result:** Camera show the image and I can scan the QR code on the other device screen

### Device info
**Device:** Galaxy Tab A7 STM-T505
**Android version:** 10
**Signal version:** 5.3.12

**Comment by Elimartain** [comment]
I have seen this same problem with my device redmi note 7 pro running on Android 11 custom rom.. 

This bug can be reproduced if u open a app (any app) and go to it's qr scan camera. I encountered this with discord and tested with another app . I got the same camera freeze with one second of working...
Hope I get a fix soon

**Comment by Erikzzz** [comment]
What solved it for me: in the android signal app go to settings -> connected devices -> add device (the '+' sign), then the signal QR scanner works. 
what did not work for me: using the build-in QR scanner (or any 3th party alternative for that matter I presume) to scan the signal desktop QR which then opens the signal app on android (via some URL) - it seems this generated url leads to a broken version of the signal android app QR scanner. 

so in short: only use the signal QR scanner, no 3th party to interim open the app. for now.

**Comment by doxioxi** [comment]
Updating Android to 11 solves this for me...

**Comment by wzurborg** [comment]
I am having this problem on a new Redmi Note 11, MIUI 13.0.1, Android 11. Please re-open the issue.

Camera freezes 0.5 sec after pressing the + button to link a device. Other scanner apps can scan the code but redirect to Signal, so I am stuck.

**Comment by wzurborg** [comment]
> What solved it for me: In the android signal app go to settings -> connected devices -> add device (the '+' sign), then the signal QR scanner works.

This is not a solution to the original problem. The camera freezes right after pressing the + button in the Signal app.

**Comment by fabian727** [comment]
same on the mobile device of a friend of mine. I am not sure, bug I thought it was a redmi note 11 too.

**Comment by cody-signal** [comment]
Please move reports to this ticket: https://github.com/signalapp/Signal-Android/issues/12009

**Comment by Skyler469** [comment]
This happens to me on iod 26.2.
